src/main/java/com/acquia/http/HMACFilter.java

```diff
@@ -25,89 +22,52 @@
 public abstract class HMACFilter implements Filter {
 
     /**
-     * The config parameter that defines a comma-separated list of custom HTTP headers 
-     * that are used in the construction of the message that will be encrypted.
-     */
-    public static final String FILTER_CONFIG_CUSTOMER_HEADERS = "customHeaders";
-    
-    /**
      * The config parameter that defines the name of the algorithm used to create the HMAC.
      */
     public static final String FILTER_CONFIG_ALGORITHM = "algorithm";
-    
+
     /**
      * The Algorithm used to create the HMAC.
      */
     HMACAlgorithm algorithm;
-    
-    /**
-     * The list of custom HTTP headers used to construct the message that will be encrypted.
-     */
-    List<String> customHeaders;
-    
+
     @Override
     public void init(FilterConfig config) throws ServletException {
-        String customHeadersList = config.getInitParameter(FILTER_CONFIG_CUSTOMER_HEADERS);
-        if ( customHeadersList != null ) {
-            this.customHeaders = new ArrayList<String>( Arrays.asList( customHeadersList.split(",") ) );
-        }
-        else {
-            this.customHeaders = new ArrayList<String>();
-        }
         HMACAlgorithmFactory algorithmFactory = new HMACAlgorithmFactory();
         String algorithmName = config.getInitParameter(FILTER_CONFIG_ALGORITHM);
         this.algorithm = algorithmFactory.createAlgorithm(algorithmName);
     }
-    
 
     @Override
     public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
             throws IOException, ServletException {
-        if ( req instanceof HttpServletRequest ) {
-            HttpServletRequest request   = (HttpServletRequest) req;
+        if (req instanceof HttpServletRequest) {
+            HttpServletRequest request = (HttpServletRequest) req;
             HttpServletResponse response = (HttpServletResponse) res;
-            
-            String authHeader = request.getHeader("Authorization");
 
+            String authHeader = request.getHeader("Authorization");
             if (authHeader != null) {
+                Map<String, String> authorizationParameterMap = HMACUtil.convertAuthorizationIntoParameterMap(authHeader);
```

Comment:
Instead of returning a map, can we create a class called HMACAuthorizationHeader that has explicit properties and methods to convert to and from a String